src/hammer-vlsi/defaults.yml

```diff
@@ -341,15 +341,44 @@ vlsi.inputs:
     #   * as wildcards in the pin names. So "*" for all pins or "io_tl_a*" for all pins that
     #   start with "io_tl_a".
     # - side (Optional[str]) - The side of the chip/block these pins will be placed along.
-    #   One of "left", "right", "top", "bottom"
+    #   One of:
+    #   - left, right, top, bottom
+    #   - internal: creates an internal pin with `location` as its *center*.
+    #     Requires the semi_auto pin generation mode.
+    #   If left unspecified, the CAD tool may choose an arbitrary side to use (likely the closest side).
     # - layers (Optional[List[str]]) - The metal layers that these pins will be placed on.
     # - preplaced (Optional[bool]) - If true, these pins are preplaced by some internal block or macro.
     #   This is telling the tool that these pins are on an internal block and I want them
     #   "placed" where they are currently on the internal block. This can be used for IP
     #   that directly connects to bumps, or signals that will be connected by abutment,
     #   and much more.
+    # - location (Optional[Tuple[float, float]]) - specifies the *center* (x, y) location of the pin on the specified
+    #   edge.
+    #   If side is one of left/right/top/bottom, this specifies the point that lies on the edge that is used for
+    #   width/depth control. Below is an example of location if side is "top":
+    #
+    #   -------X------- (chip/block edge)
+    #   |             |
+    #   ---------------
+    #
+    #   If side is "internal", then this specifies the absolute *center* of the
+    #   Requires the semi_auto pin generation mode.
+    # - width (Optional[float]) - specifies a custom width for the pin. The width direction is orthogonal to the routing
+    #   direction.
+    #   Requires the semi_auto pin generation mode.
+    # - depth (Optional[float]) - specifies a custom depth for the pin. The depth direction is parallel to the routing
+    #   direction.
+    #   Requires the semi_auto pin generation mode.
     assignments: []
 
+    # Pin generation modes.
```

Comment:
Can we move this above the assignments struct so the references to it aren't backwards?